### PR TITLE
Add endpoint description field to API spec

### PR DESCRIPTION
**Pull request by wkmor1**
With this change the first line of endpoint block comments are interpreted as OpenAPI 'summary' field and subsequent comment lines interpreted as the 'description' field. Technically this is a breaking change for anyone who is intentionally breaking 'summary' field comments over more than one line. On the other hand maybe more intuitive as splitting line 1 vs line 2-N is more like the behavior of Roxygen comments -> DESCRIPTION file fields, title and description.

### CODE REVIEW

File: R/plumb-block.R [hunk 1]
```diff
@@ -250,7 +250,8 @@ plumbBlock <- function(lineNum, file, envir = parent.frame()){
     parsers = rev(parsers),
     assets = assets,
     params = rev(params),
-    comments = paste0(rev(comments), collapse = " "),
+    comments = paste0(rev(comments)[1], collapse = " "),
+    description = paste0(rev(comments)[-1], collapse = " "),
```
Comment (schloerke) [inline R/plumb-block.R:254]: Should we collapse using a new line? Feels odd to have two multi-line paragraphs result into a large, single paragraph.

```suggestion
    description = paste0(rev(comments)[-1], collapse = "\n"),
```

File: NEWS.md [hunk 2]
```diff
@@ -2,7 +2,9 @@
 
 ## Breaking changes
 
-## New features
+* First line of endpoint comments interpreted as OpenAPI 'summary' field and subsequent comment lines interpreted as 'description' field.
```
Comment (schloerke) [inline NEWS.md:5]:
```suggestion
* First line of endpoint comments interpreted as OpenAPI 'summary' field and subsequent comment lines interpreted as 'description' field. (@wkmor1 #805)
```

File: R/plumb-block.R (hunk 1 above)
Comment (wkmor1) [reply R/plumb-block.R:254]: Makes sense

File: R/plumb-block.R [hunk 3]
```diff
@@ -250,7 +250,8 @@ plumbBlock <- function(lineNum, file, envir = parent.frame()){
     parsers = rev(parsers),
     assets = assets,
     params = rev(params),
-    comments = paste0(rev(comments), collapse = " "),
+    comments = paste0(rev(comments)[1], collapse = " "),
```
[the remark this answers is not in the document]
Comment (wkmor1) [reply R/plumb-block.R:253]: Probs not

File: tests/testthat/test-parse-block.R [hunk 4]
```diff
@@ -281,10 +284,11 @@ test_that("block respect original order of lines for comments, tags and response
     "#' @tag bbb",
     "#' comments first line",
     "#' comments second line",
+    "#' comments third line",
     "#' @response 200 ok",
     "#' @response 404 not ok")
   b <- plumbBlock(length(lines), lines)
-  expect_equal(b$comments, "comments first line comments second line")
+  expect_equal(b$description, "comments second line\ncomments third line")
```
Comment (schloerke) [inline tests/testthat/test-parse-block.R:291]:
```suggestion
  expect_equal(b$comments, "comments first line")
  expect_equal(b$description, "comments second line\ncomments third line")
```

**Review by schloerke — APPROVED** [review]

**Comment by schloerke** [comment]
Thank you! (Sorry clicked the wrong button there)

**Pull request merged**